digitizer/parse.py

```diff
@@ -52,12 +53,22 @@ def prepare_isotherm_dict(form):
     data['isotherm_type'] = form.inp_isotherm_type.value
     data['measurement_type'] = form.inp_measurement_type.value
 
-    measurements = re.sub(' +', ' ', form.inp_isotherm_data.value)
-    measurements = measurements.replace(' ', ',')
-    measurements = np.genfromtxt(StringIO(measurements),
-                                 delimiter=',',
-                                 comments='#')
-    measurements = np.array(measurements, ndmin=2)  # deal with single data row
+    measurements = form.inp_isotherm_data.value  # get string from input text box
+    for delimiter in ['\t', ';', '|', ',']:
+        measurements = measurements.replace(
+            delimiter, ' ')  # convert all delimiters to spaces
+    measurements = re.sub(' +', ' ', measurements)  # collapse whitespace
+    print(measurements)
+    measurements = pd.read_table(
+        StringIO(measurements),
+        sep=',| ',
+        #sep=' ',
+        # for some reason, leaving only the space delimiter is causing a problem
+        #  when lines have trailing whitespace. Need to check pandas documentation
```

Comment:
I see. Alternatively, we could iterate over the lines and `strip` them

Reply:
That was the inelegant solution that came to mind to me as well. Will still check pandas docs